### DISCUSSION TITLE
How to close a section in x-show

### DISCUSSION THREAD

**Discussion by sqllyw** [alpinejs/alpine#1076] [1. Help] [answered]
Hi,
I got a code like this, 

https://jsfiddle.net/z9afmtb1/

after I click the submit button, it calls a function add_to_cart, inside that function I set open = false, but the div under x-show='open' is still visible, any idea why? Thanks

**Comment by KevinBatdorf** [comment, marked as answer]
You need to put your methods on the component directly, like this:

```js
function get_data() {
  return {
    open: false,
    remarks: 'r',
    add_to_cart(remark) {        
      console.log(remark)
      this.open = false
    }
  }
}
```
Also, note that you use `this.open` instead of `open` there.

**Reply by sqllyw** [reply]
it works, thanks
